src/app/modules/core/components/map/map.component.ts

```diff
@@ -429,22 +463,38 @@ export class MapComponent extends deepCopy implements OnInit, AfterViewInit, OnC
     }
   }
 
-  private formatReaches(data): any {
-    const streamArray = [];
-    for (let i = 0; i < data.features.length; i++) {
-      if (data.features[i].geometry.type === 'LineString') {
-        const polylinePoints = this.deepCopy(data.features[i]);
-        streamArray.push(polylinePoints);
+  private wm(msg: string, title?: string, timeout?: number) {
+    try {
+      let options: Partial<IndividualConfig> = null;
+      if (timeout) { options = { timeOut: timeout }; }
+      if (timeout == 0) {
+        options = {
+          disableTimeOut : true,
+          timeOut: 0,
+          extendedTimeOut: 0,
+          tapToDismiss: false
+        };
       }
+      this.messager.warning(msg, title, options);
+    } catch (e) {
     }
-    streamArray.map((reach) => {
-      reach.properties.show = false;
-    });
+  }
```

Comment:
I may be doing something wrong but prior to setting up the two separate functions I couldn't get the colors of the message boxes to change by passing a messageType. It's why I went this route because by default the warning is orange and the error is red.

Reply:
I maybe explained poorly - I meant to send a message type to your messaging function, not to the messager. So you would make a function something like `private nm(msg: string, messageType: string, title?: string, timeout?: number)`, and at the end, you would treat it with that, so you would say `if (messageType == 'warning') this.messager.warning(msg, title, options)`, and same if it's an error. The only difference would be the actual function called (`this.messager.warning` v. `this.messager.error` or whatever it is)